pipewire/qubes-pw-module.c

```diff
@@ -566,8 +628,25 @@ static int add_stream(struct spa_loop *loop,
 static int connect_stream(struct qubes_stream *stream)
 {
     const char *msg = qubes_stream_is_capture(stream) ? "capture" : "playback";
-    uint16_t domid = stream->impl->domid;
+    int domid = -EINVAL, domid_from_props, domid_from_qdb;
     int status;
+    
+    domid_from_props = get_domid_from_props(stream->impl->props);
+    domid_from_qdb = get_domid_from_qdb(stream->impl->qdb);
+    if (domid_from_props>=0) {
+        pw_log_debug("using domid from pw props");
+        domid = domid_from_props;
+    } else if (domid_from_qdb>=0) {
+        pw_log_debug("using domid from qdb");
+        domid = domid_from_qdb;
+    }
+    
+    if (domid<0) {
+        pw_log_debug("no domid, cannot create stream");
+        return 0;
```

Comment:
This should be a negative number.

Reply:
At first, I thought returning negative value would exit main loop, but it looks like it is not the case. I've adjusted.